### ISSUE TITLE
Add support for Date only (ie w/o time) data type

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#533]
_Original author: danpaulsmith (February 14, 2012 15:03:47)_

When gathering user feedback on our extension for Refine the other day, a W3C member mentioned that when formatting a date (e.g. 07/05/2011) - information is created which isn't correct.

A formatted date gains the &quot;00:00:00Z&quot; (e.g. 2011-05-07T00:00:00Z) - stating that whatever this date is - happened at midnight - which is actually untrue.

_Original issue: http://code.google.com/p/google-refine/issues/detail?id=533_

**Comment by thadguidry** [comment]
Java 8+ now has a class to store Dates without time or timezone (value-based, needs `equals` for comparisons).
https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/time/LocalDate.html

**Comment by tfmorris** [comment]
The Java 8 date implementation is basically the same as the Joda package that I mentioned here: https://github.com/OpenRefine/OpenRefine/issues/533#issuecomment-9433294

**Comment by thadguidry** [comment]
@tfmorris Yes, that I knew about.  (I still love Moment.js ways of working a bit more however, but our GREL is a bit better, but could be better still and closer to Moment's style of handling)

**Comment by tfmorris** [comment]
I still believe that OpenRefine would benefit greatly from a Date datatype (and perhaps a Time datatype too).

Although what I said in 2012 about the internal format is true, what the user sees in Excel is the formatted version, so that's what's important to them. Even though they are all just floating point numbers internally, if the user sees it as a formatted date, it's a date to them.

Additionally, the original request didn't have anything to do with Excel at all.

**Comment by ostephens** [comment]
>I still believe that OpenRefine would benefit greatly from a Date datatype (and perhaps a Time datatype too).

Completely agree. In library and archive data (which I work with, and train people to work with), there are a lot of dates without times and converting them to date times adds a spurious degree of specificity. The date information I encounter can be stated at year, year-month or year-month-day depending. To give a few examples from libraries:

* Books have a publication year. Representing these as a date in OR currently means choosing a specific date and time in that year
* Journal/serial issues can be dated by year-month (e.g. the December 2023 issue of a particular publication)
* Journal issues or specific articles might have a date of publication specified as year-month-day, but not a specific time

Currently the workflow needed tends to be like:`

**Comment by emcaulay** [comment]
I also would like some way to support lack-of-granularity in date data. I am also a librarian and archives-related professional. 

Right now, I'm trying to convert string data reflecting dates from one format to another. (e.g., instead of 01-1978, I want "1978-01").
